subiquity/models/filesystem.py

```diff
@@ -218,36 +218,63 @@ def dehumanize_size(size):
     return num * mult // div
 
 
-def round_raid_size(min_size):
-    # The calculation of how much of a device mdadm uses for raid is a
-    # touch ridiculous. What follows is a translation of the code at:
-    # https://git.kernel.org/pub/scm/utils/mdadm/mdadm.git/tree/super1.c?h=mdadm-4.1&id=20e8fe52e7190b3ffda127566852eac2eb7fa1f7#n2770
-    # (note that that calculation is in terms of 512-byte sectors and
-    # this one is in bytes).
-    #
-    # This makes assumptions about the defaults mdadm uses but mostly
-    # that the default metadata version is 1.2, and other formats use
-    # less space.
-    bmspace = 128*1024
-    headroom = 128*1024*1024
-    while (headroom << 10) > min_size and headroom > 2*1024*1024:
+DEFAULT_CHUNK = 512
+
+
+# The calculation of how much of a device mdadm uses for raid is more than a
+# touch ridiculous. What follows is a translation of the code at:
+# https://git.kernel.org/pub/scm/utils/mdadm/mdadm.git/tree/super1.c,
+# specifically choose_bm_space and the end of validate_geometry1. Note that
+# that calculations are in terms of 512-byte sectors.
+#
+# We make some assumptions about the defaults mdadm uses but mostly that the
+# default metadata version is 1.2, and other formats use less space.
+#
+# Note that data_offset is computed for the first disk mdadm examines and then
+# used for all devices, so the order matters! (Well, if the size of the
+# devices vary, which is not normal but also not something we prevent).
+#
+# All this is tested against reality in ./scripts/get-raid-sizes.py
+def calculate_data_offset(devsize):
+    devsize >>= 9  # convert to sectors
```

Comment:
I think we could add a comment here to the effect that we're using sectors to make it easier to match the kernel mdadm algorithm and we convert back to bytes.

I'd also suggest renaming this function to 'calculate_data_offset_bytes'

Reply:
Fair on both counts.